pf/tfbridge/provider_read.go

```diff
@@ -90,12 +90,35 @@ func (p *provider) ReadWithContext(
 		}
 
 		// __defaults is not needed for Plugin Framework bridged providers
-		delete(result.Inputs, "__defaults")
+		deleteDefaultsKey(&result.Inputs)
 	}
 
 	return result, ignoredStatus, err
 }
 
+// deleteDefaultsKey removes the `__defaults: []` entry from all objects recursively
+// The `__defaults` key is something used in sdkv2 and is not handled here in pf. Because
+// of some code reuse between sdkv2 & pf the `__defaults` key is getting inserted
+func deleteDefaultsKey(inputs *resource.PropertyMap) {
```

Comment:
Have a look at propertyvalue.TransformPropertyValue that also goes under secrets etc and handles the recursion hopefully properly without any further bugs. Might be easier to review, we're just transforming everything and we're dropping __defaults whenever we find them.

Reply:
Done!